contents/Documentation/Patterns.playground/Pages/Monad comprehensions.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,264 @@
+// nef:begin:header
+/*
+ layout: docs
+ title: Monad comprehensions
+ */
+// nef:end
+// nef:begin:hidden
+import Foundation
+import Bow
+import BowEffects
+// nef:end
+/*:
+ # Monad comprehensions
+ 
+ {:.beginner}
+ beginner
+ 
+ Monad comprehensions are a generalization over the syntax provided in Swift to unwrap `Optional`s using `if let` or `guard let`, or to safely run throwing functions using `try?`. This page shows how Bow provides an uniform syntax to have direct, imperative-like style of working with monadic values.
+ 
+ ## Motivation
+ 
+ For instance, consider the following function receiving three optional values of types `Int`, `Double` and `String`. The purpose of the function is to unwrap the three values and make a `String` with the contents of all three, or return `nil` if any of them does not have a value. It can be implemented using the `if let` syntax to safely unwrap the optionals:
+ */
+func join_ifLet(_ a: Int?, _ b: Double?, _ c: String?) -> String? {
+    if let x = a,
+       let y = b,
+       let z = c {
+        return "\(x), \(y), \(z)"
+    }
+    return nil
+}
+/*:
+ This is equivalent to the following implementation:
+ */
+func join_flatMap(_ a: Int?, _ b: Double?, _ c: String?) -> String? {
+    return a.flatMap { x in
+        b.flatMap { y in
```

Comment:
could you add indentation to `flatMap`? to show how it starts looking like a pyramid